sharding-and-replication.md

```diff
@@ -0,0 +1,171 @@
+the storage.
+
+## Sharding
+
+Sharding is a feature that allows splitting the data into multiple Clickhouse nodes to
+increase throughput and decrease latency.
+The sharding feature uses `Distributed` engine that is backed by local tables.
+The distributed engine is a "virtual" table that does not store any data. It is used as 
+an interface to insert and query data.
+
+To setup sharding run the following statements on all nodes in the cluster.
+The "local" tables have to be created on the nodes before the distributed table.
+
+```sql
+CREATE TABLE IF NOT EXISTS jaeger_spans AS jaeger_spans_local ENGINE = Distributed('{cluster}', default, jaeger_spans_local, cityHash64(traceID));
+CREATE TABLE IF NOT EXISTS jaeger_index AS jaeger_index_local ENGINE = Distributed('{cluster}', default, jaeger_index_local, cityHash64(traceID));
+CREATE TABLE IF NOT EXISTS jaeger_operations AS jaeger_operations_local ENGINE = Distributed('{cluster}', default, jaeger_operations_local, rand());
+```
+
+* The `AS <table-name>` statement creates table with the same schema as the specified one. 
+* The `Distributed` engine takes as parameters cluster , database, table name and sharding key.
+
+If the distributed table is not created on all Clickhouse nodes the Jaeger query fails to get the data from the storage.
+
+### Deploy Clickhouse
+
+Deploy Clickhouse with 2 shards:
+
+```yaml
+cat <<EOF | kubectl apply -f -
+apiVersion: "clickhouse.altinity.com/v1"
+kind: "ClickHouseInstallation"
+metadata:
+  name: jaeger
+spec:
+  configuration:
+    clusters:
+      - name: cluster1
+        layout:
+          shardsCount: 2
+EOF
+```
+
+Use the following command to run `clickhouse-client` on Clickhouse nodes and create the distributed tables:
+```bash
+kubectl exec -it statefulset.apps/chi-jaeger-cluster1-0-0 -- clickhouse-client  
+```
+
+### Plugin configuration
+
+The plugin has to be configured to write and read that from the global tables:
+
+```yaml
+address: tcp://clickhouse-jaeger:9000
+spans_table: jaeger_spans
+spans_index_table: jaeger_index
+operations_table: jaeger_operations
+```
+
+## Replication
+
+Replication as the name suggest automatically replicates the data across multiple Clickhouse nodes.
+It is used to accomplish high availability, load scaling and migration/updates.
+
+The replication uses Zookeeper. Refer to the Clickhouse operator how to deploy Zookeeper.
+
+Zookeeper allows us to use `ON CLUSTER` to automatically replicate table creation on all nodes.
+Therefore the following command can be run only on a single Clickhouse node:
+
+```sql
+CREATE TABLE IF NOT EXISTS jaeger_spans_local ON CLUSTER '{cluster}'  (
+                                                                timestamp DateTime CODEC(Delta, ZSTD(1)),
+    traceID String CODEC(ZSTD(1)),
+    model String CODEC(ZSTD(3))
+    ) ENGINE ReplicatedMergeTree('/clickhouse/tables/{shard}/jaeger_spans', '{replica}')
```

Comment:
Perhaps it might make sense to omit ReplicatedMergeTree parameters at all, as per email thread discussion: Atomic DB engine will choose a path automatically so there wont be any conflicts when cluster nodes are re-created.

Reply:
Done here https://github.com/pavolloffay/jaeger-clickhouse/pull/49